### PR TITLE
Issue 9723: document default Vim server name

### PR DESCRIPTION
**Pull request by ghost**
This pull request documents the existence and use of a default server name when using the builtin functions remote_expr, remote_foreground and remote_send.

### CODE REVIEW

**Comment by brammool** [comment]
Thanks.   It's not always GVIM though, it depends on how Vim was started.  This is explained in the "SERVER NAME" section in remote.txt.  I'll add a paragraph there and link to it, so that there is no repetition.

**Comment by msoyka-of-wharton** [comment]
Bram,

I based my changes on the code in function "serverSendToVim", line 389.  That function is ultimately called from f_remote_expr and the other two so I don't see how the "default name" could be anything other than "GVIM".  How Vim matches the name with a server later is a separate issue.

I understand that a server can have any name but the issue I was addressing involved the client-side using an empty string for the server name in the three builtins.  Is there something more that I'm missing?

-mike

**Comment by brammool** [comment]
> I based my changes on the code in function "serverSendToVim", line
> 389.  That function is ultimately called from f_remote_expr and the
> other two so I don't see how the "default name" could be anything
> other than "GVIM".  How Vim matches the name with a server later is a
> separate issue.
> 
> I understand that a server can have any name but the issue I was
> addressing involved the client-side using an empty string for the
> server name in the three builtins.  Is there something more that I'm
> missing?

Ah yes, in the case of using the X server "GVIM" is hard coded.

For MS-Windows I can't actually see what it will use.  I guess an empty
name won't work.

-- 
hundred-and-one symptoms of being an internet addict:
43. You tell the kids they can't use the computer because "Daddy's got work to
    do" and you don't even have a job.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by msoyka-of-wharton** [comment]
On Tue, Feb 15, 2022 at 11:18 AM Bram Moolenaar ***@***.***>
wrote:

>
> > I based my changes on the code in function "serverSendToVim", line
> > 389. That function is ultimately called from f_remote_expr and the
> > other two so I don't see how the "default name" could be anything
> > other than "GVIM". How Vim matches the name with a server later is a
> > separate issue.
> >
> > I understand that a server can have any name but the issue I was
> > addressing involved the client-side using an empty string for the
> > server name in the three builtins. Is there something more that I'm
> > missing?
>
> Ah yes, in the case of using the X server "GVIM" is hard coded.
>
> For MS-Windows I can't actually see what it will use. I guess an empty
> name won't work.
>

Hmm, I guess I didn't look that closely so I put it to the test.
Running vim 8.2.3027 under Windows 10 with and without the presence of a
server named GVIM:

  remote_expr("","") fails with error E241,
  remote_send("","") fails with error E241,
  remote_foreground("") does not fail and appears to do nothing.

So Windows and X11 systems behave differently when those functions are
passed an empty string for a servername.

In summary, without a GVIM server running:
  remote_expr("","") fails with error E241 for both Windows 10 and X11,
  remote_send("","") fails with error E241 for both Windows 10 and X11,
  remote_foreground("") fails with E241 for X11 but does not fail for
Windows 10.

-mike
